lib/modules/datasource/go/base.ts

```diff
@@ -154,17 +155,35 @@ export class BaseGoDatasource {
       const parsedUrl = URL.parse(goSourceUrl);
 
       // TODO: `parsedUrl.pathname` can be undefined
-      const packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
+      let packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
 
-      const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
+      const endpoint = GlobalConfig.get('endpoint');
+
+      const endpointPrefix = endpoint?.match('https://[^/]*/(.*)api/v4/?');
```

Comment:
`api/v4` should be optional, we will soon remove it from endpoint on platform init.

Reply:
@viceice I changed the regular expression and extended the test case. It is now optional.